tests/scripts/thread-cert/test_netdata_publisher.py

```diff
@@ -0,0 +1,307 @@
+        # DNS/SRP unicast entries
+
+        # Publish DNS/SRP unicast address on all routers, first using
+        # ML-EID address, then change to use specific address. Verify
+        # that number of entries in network data is correct in each step
+        # and that entries are switched correctly.
+
+        num = 0
+        for node in routers:
+            node.netdata_publish_dnssrp_unicast_mleid(DNSSRP_PORT)
+            self.simulator.go(WAIT_TIME)
+            num += 1
+            services = leader.get_services()
+            self.assertEqual(len(services), min(num, DESIRED_NUM_DNSSRP_UNCIAST))
+            self.verify_unicast_services(services)
+
+        for node in routers:
+            node.netdata_publish_dnssrp_unicast(DNSSRP_ADDRESS, DNSSRP_PORT)
+            self.simulator.go(WAIT_TIME)
+            services = leader.get_services()
+            self.assertEqual(len(services), min(num, DESIRED_NUM_DNSSRP_UNCIAST))
+            self.verify_unicast_services(services)
+
+        for node in routers:
+            node.netdata_unpublish_dnssrp()
+            self.simulator.go(WAIT_TIME)
+            num -= 1
+            services = leader.get_services()
+            self.assertEqual(len(services), min(num, DESIRED_NUM_DNSSRP_UNCIAST))
+            self.verify_unicast_services(services)
+
+        #---------------------------------------------------------------------------------
+        # Verify publisher preference when removing entries.
+        #
+        # Publish DNS/SRP anycast on 8 nodes: leader, router1,
+        # router2, and all 5 end-devices. Afterwards, manually add
+        # the same service entry in Network Data on router3, router4,
+        # and router5 and at each step check that entry from one of
+        # the end-devices is removed (publisher prefers
+        # entries from routers over the ones from end-devices).
+
+        num = 0
+        for node in [leader, router1, router2] + end_devs:
+            node.netdata_publish_dnssrp_anycast(ANYCAST_SEQ_NUM)
+            self.simulator.go(WAIT_TIME)
+            num += 1
+            services = leader.get_services()
+            self.assertEqual(len(services), num)
+            self.verify_anycast_services(services)
+
+        self.assertEqual(num, DESIRED_NUM_DNSSRP_ANYCAST)
+
+        service_data = '%02x%02x' % (ANYCAST_SERVICE_NUM, int(ANYCAST_SEQ_NUM))
+
+        for node in [router3, router4, router5]:
+            node.add_service(str(THREAD_ENTERPRISE_NUMBER), service_data, '00')
+            node.register_netdata()
+            self.simulator.go(WAIT_TIME)
+
+            services = leader.get_services()
+            self.assertEqual(len(services), num)
+            self.verify_anycast_services(services)
+
+            service_rlocs = [int(service[4], 16) for service in services]
+
+            for node in [leader, router1, router2]:
+                self.assertIn(node.get_addr16(), service_rlocs)
```

Comment:
Check that the router which has newly registered its service is in the Thread Network Data.

```suggestion
            for server in [leader, router1, router2,  node]:
                self.assertIn(server.get_addr16(), service_rlocs)
```

Reply:
Changed in new push.